espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/UpdateSettings.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.espressoserver.lib.handlers
+
+import io.appium.espressoserver.lib.handlers.exceptions.AppiumException
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidArgumentException
+import io.appium.espressoserver.lib.helpers.AndroidLogger
+import io.appium.espressoserver.lib.model.SettingsParams
+import io.appium.espressoserver.lib.model.settings.AbstractSetting
+import io.appium.espressoserver.lib.model.settings.SettingType
+
+class UpdateSettings : RequestHandler<SettingsParams, Void?> {
+    override fun handleInternal(params: SettingsParams): Void? {
+        params.settings?.let {
+            for ((settingName: String, settingValue: Any?) in it.entries) {
+                AndroidLogger.debug("$settingName $settingValue")
+                val setting = getSetting(settingName)
+                setting.apply(settingValue)
+            }
+        }
+        return null
+    }
+
+    private fun getSetting(settingName: String): AbstractSetting {
+        for (settingType: SettingType in SettingType.values()) {
```

Comment:
this could be reduced to 
```
private fun getSetting(settingName: String): AbstractSetting =
  SettingType.values().find { it.name == settingName } :? throw InvalidArgumentException()
```

Reply:
changed